modules/rs_grading.py

```diff
@@ -664,3 +683,70 @@ def _score_practice_quality(practice_start_time, course_name, sid, points, score
         else:
             practice_score = 1
     return (practice_score, trials_count)
+
+
+def do_fill_user_topic_practice_log_missings(db, settings):
```

Comment:
This is going to be very inefficient. For every log record, you're rerunning the query that gets all the flashcards for the user. (And I think lines 697-706 are duplicated right below it, and more stuff below that; please check that whole section of code.)

It's OK if it's inefficient since we're only ever going to run this code once. But we should run it from the rsmanage script rather than the web interface. See rsmanage/rsmanager.py line 359 and the file rsmanage/grade.py for a template for how to do this. I implemented those in order to be able to run the slow lecture prep grading commands.

Reply:
Now, we can run it through the CLI.
In terms of the efficiency, I did not spend any time trying to make it more efficient. I think we only need to use it once and no one else is going to use it.
I added some comments explaining different parts of the code. I don't see any duplicates!